app/Models/Mship/Concerns/HasNotifications.php

```diff
@@ -22,6 +22,15 @@ public function readSystemNotifications()
             ->withPivot(['created_at']);
     }
 
+    /**
+     * @param Notification $notification
+     * @return bool
+     */
+    public function hasReadNotification(SysNotification $notification)
```

Comment:
It would be good to add some assertions in the tests about this relationship.

e.g.

`$this->assertFalse($this->account->hasReadNotification($notification);`

`$this->post......`

`$this->assertTrue($this->account->hasReadNotification($notification);`

It may feel more natural for this to be done at unit level.